src/main/java/com/whichclasses/frontend/SearchApiServlet.java

```diff
@@ -29,8 +35,31 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
     response.setStatus(HttpServletResponse.SC_OK);
     response.setContentType("application/json");
-    // TODO: actually implement this
-    response.getWriter().write("{\"results\":[{\"identifier\":{\"department\":\"ECE\",\"course_number\":\"351A\"},\"title\":\"All The Circuits\"}]}");
+    String quality = request.getParameter("quality");
+    String entityType = request.getParameter("entityType");
+    String department = request.getParameter("department");
+    
+    Question question = Question.OVERALL_COURSE_RATING;
+    switch(quality) {
```

Comment:
Actually, this whole switch statement should just be a Map<String, Question>. If you don't find the key, return a 400 (bad request).